### DISCUSSION TITLE
Pattern Matching

### DISCUSSION THREAD

**Discussion by damodaranr** [CacheControl/json-rules-engine#262] [Ideas]
Could we get regex pattern matching for strings please?

I did extend the engine-default-operator.js and operator.json in the associated editor project to allow this:

Operators.push(new _operator2.default('pattern', function (a, b) {
  return new RegExp(b, 'gim').test(a);
}));
Operators.push(new _operator2.default('notPattern', function (a, b) {
  return !(new RegExp(b, 'gim').test(a));
}));

**Comment by CacheControl** [comment]
@damodaranr please see the custom operators example for how to implement this: https://github.com/CacheControl/json-rules-engine/blob/master/examples/06-custom-operators.js
